ReactWindows/ReactNative.Shared/ReactInstanceManager.cs

```diff
@@ -171,23 +171,37 @@ public ReactContext CurrentReactContext
                 }
                 _hasStartedCreatingInitialContext = true;
 
-                return await CreateReactContextCoreAsync(token);
+                ReactContext context = null;
+                try
```

Comment:
why is this in "try"? The description doesn't say it will throw?

Reply:
The method throws at least when the passed in token is canceled.

---
In reply to: [186898714](https://github.com/Microsoft/react-native-windows/pull/1827#discussion_r186898714) [](ancestors = 186898714)

Reply:
@rozele it's an OnSuspend scenario done during the initialization. For some reason the instance manager cancels the init with no exception (exception is used only if caller cancels the token, a separate scenario basically) and we get back a null instance.

Now it's debatable whether the initialization has to be canceled during OnSuspend, but I didn't challenged that choice.


---
In reply to: [189277767](https://github.com/Microsoft/react-native-windows/pull/1827#discussion_r189277767) [](ancestors = 189277767)

Reply:
Now, I suspect the other scenario (caller cancels the token) is valid too, perhaps we didn't use it ourselves.
The fix covers both.

---
In reply to: [189307613](https://github.com/Microsoft/react-native-windows/pull/1827#discussion_r189307613) [](ancestors = 189307613,189277767)